frontend/packages/text-editor/src/TextEditor.tsx

```diff
@@ -43,11 +43,28 @@ export const TextEditor = ({
 }: TextEditorProps) => {
   const { t } = useTranslation();
   const resourceRows = mapResourceFilesToTableRows(textResourceFiles);
+
+  const previousSelectedLanguages = useRef<string[]>([]);
+
   const availableLangCodesFiltered = useMemo(
     () => availableLanguages?.filter((code) => ISO6391.validate(code)),
     [availableLanguages],
   );
 
+  useEffect(() => {
+    const addedLanguage = selectedLangCodes.find(
+      (lang) => !previousSelectedLanguages.current.includes(lang),
+    );
+
+    if (addedLanguage) {
+      const elementToFocus: HTMLElement = document.getElementById('header-lang' + addedLanguage);
+      if (elementToFocus) {
+        elementToFocus.scrollIntoView({ behavior: 'smooth', inline: 'center' });
+      }
+    }
+    previousSelectedLanguages.current = selectedLangCodes;
+  }, [selectedLangCodes.length]);
```

Comment:
Is it on purpose that this useEffect is missing the dependency ```selectedLangCodes```? 😄

Reply:
Sort of. I figured that the reference was not enough since the ref is still the same even though the size changes. But I guess it is more bulletproof adding the ref to the object itself also. 
Will do it!